src/main/java/at/ac/fhcampuswien/fhmdb/HomeController.java

```diff
@@ -13,104 +13,160 @@
+     */
+    public void initialize() {
+        List<Movie> allMovies;
+
+        try {
+            allMovies = MovieAPIService.getMovies();
+        } catch (IOException e){
+            allMovies = new ArrayList<>();
+        }
+
         observableMovies.addAll(allMovies);
+        sortMovies();
 
-        movieListView.setItems(filteredList);
-        movieListView.setCellFactory(movieListView -> new MovieCell());
+        movieListView.setItems(filteredMovies);
+        movieListView.setCellFactory(e -> new MovieCell());
 
-        genreComboBox.setPromptText("Filter by Genre");
         genreComboBox.getItems().addAll(Genre.values());
+        genreComboBox.setValue(Genre.NO_FILTER);
 
-        releaseYearPicker.setPromptText("Filter by Release Year");
-        releaseYearPicker.getItems().addAll(getYears());
-
-        ratingComboBox.setPromptText("Filter by Rating");
-        ratingComboBox.getItems().addAll(RatingOption.values());
-
-        searchBtn.setOnAction(actionEvent -> searchForMovie(searchField.getText().trim(), genreComboBox.getValue(), filteredList, observableMovies));
+        releaseYearPicker.getItems().addAll(Year.values());
+        releaseYearPicker.setValue(Year.NO_FILTER);
 
-        sortBtn.setOnAction(actionEvent -> {
-            if(sortBtn.getText().equals("Sort (asc)")) {
-                observableMovies.sort(Comparator.naturalOrder());
-                sortBtn.setText("Sort (desc)");
-            } else {
-                observableMovies.sort(Collections.reverseOrder());
-                sortBtn.setText("Sort (asc)");
-            }
-        });
+        ratingComboBox.getItems().addAll(Rating.values());
+        ratingComboBox.setValue(Rating.NO_FILTER);
 
-        resetFilterBtn.setOnAction(actionEvent -> resetFilterCriteria(genreComboBox, filteredList, searchField));
+        searchBtn.setOnAction(actionEvent -> setFilter());
+        sortBtn.setOnAction(actionEvent -> sortMoviesAndSetButtonText());
+        resetFilterBtn.setOnAction(actionEvent -> resetFilter());
     }
 
-    public static void resetFilterCriteria(JFXComboBox<Genre> genreComboBox,
-                                           FilteredList<Movie> filteredList,
-                                           TextField searchField) {
+    private void resetFilter() {
+        observableMovies.clear();
         searchField.clear();
-        genreComboBox.setValue(Genre.ALL);
-        if (genreComboBox.getValue() != null) {
-            genreComboBox.getSelectionModel().clearSelection();
+
+        sortBtn.setText(SORT_DEFAULT_TEXT_DESC);
+
+        genreComboBox.setValue(Genre.NO_FILTER);
+        releaseYearPicker.setValue(Year.NO_FILTER);
+        ratingComboBox.setValue(Rating.NO_FILTER);
+
+        List<Movie> allMovies;
+
+        try {
+            allMovies = MovieAPIService.getMovies();
+        } catch (IOException e){
+            allMovies = new ArrayList<>();
```

Comment:
Duplicate code with lines 88-92. I would extract this in a method

Reply:
Extracted into private method.